### PR TITLE
feat(css/lints): Implement `no-duplicate-at-import-rules` rule

### PR DESCRIPTION
**Pull request by g-plane**
**Description:**

Implemented [`no-duplicate-at-import-rules`](https://stylelint.io/user-guide/rules/list/no-duplicate-at-import-rules/).

**BREAKING CHANGE:**

No.

### CODE REVIEW

**Review by alexander-akait — COMMENTED** [review]
Let's add tests with layer https://www.w3.org/TR/css-cascade-5/#at-import

**Comment by g-plane** [comment]
Are we missing anything? Why do we mark this PR "work in progress"?

**Comment by kdy1** [comment]
Because of https://github.com/swc-project/swc/pull/3833#pullrequestreview-899319493

I marked it as a draft (WIP) because I don't want to poll PRs

**Comment by g-plane** [comment]
@alexander-akait Are there any differences for the code below?

```css
@import "url.css" layer;
```

```css
@import "url.css" layer();
```

```css
@import "url.css" layer(default);
```

**Comment by g-plane** [comment]
@alexander-akait Can you answer the question above? Thanks.

**Comment by alexander-akait** [comment]
@g-plane Sorry for delay:
```css
@import "url.css" layer;
@import "url.css" layer();
```

are the same, but

```css
@import "url.css" layer(default);
@import "url.css" layer(other);
```

Are not, for example you can have:
```
@import "normalize.css" layer(framework);
@import "normalize.css" layer(my-library);
```

But in other hand our rule about duplication in URLs, and maybe we should report here anyway, that's a good question for discussion, currently I think we should always report about duplications, maybe we will revisit it late (will be great to open an issue in stylelint too, so that we can decide on a more correct behavior).

So right now let's add tests to ensure we always report about duplication with layer 

https://www.w3.org/TR/css-cascade-5/#cascading

**Review by alexander-akait — DISMISSED** [review]

**Comment by g-plane** [comment]
@kdy1 This can be reviewed now.

File: crates/swc_css_lints/src/rules/no_duplicate_at_import_rules.rs [hunk 1]
```diff
@@ -0,0 +1,78 @@
+use std::fmt::Display;
+
+use ahash::AHashSet;
```
Comment (kdy1) [inline crates/swc_css_lints/src/rules/no_duplicate_at_import_rules.rs:3]: Please use `AHashSet` from `swc_common` instead.

Comment (kdy1) [reply crates/swc_css_lints/src/rules/no_duplicate_at_import_rules.rs:3]: (For consisstency.)

Comment (kdy1) [reply crates/swc_css_lints/src/rules/no_duplicate_at_import_rules.rs:3]: `ahash::AHashSet` and `ahash::AHashMap` are newtype structs, while those in `swc_common` is type alias for standard hashmap type.

**Review by kdy1 — CHANGES_REQUESTED** [review]
Thank you!
I reviewed only rust code and I'll request review for css parts, as I don't know well about css

**Review by alexander-akait — APPROVED** [review]

**Review by kdy1 — APPROVED** [review]
Thank you!

---

swc-bump:
 - swc_css_lints

**Review by kdy1 — CHANGES_REQUESTED** [review]
CI failed

**Review by kdy1 — APPROVED** [review]

**Comment by kdy1** [comment]
Can you rebase?

**Review by kdy1 — CHANGES_REQUESTED** [review]
There are some merge conflicts

**Review by kdy1 — APPROVED** [review]
LGTM. Thanks!

---

swc-bump:
 - swc_css_lints

**Comment by yisibl** [comment]
> @alexander-akait Are there any differences for the code below?
> 
> ```css
> @import "url.css" layer;
> ```
> 
> ```css
> @import "url.css" layer();
> ```
> 
> ```css
> @import "url.css" layer(default);
> ```

`@import "url.css" layer();` is a wrong syntax. see https://github.com/w3c/csswg-drafts/pull/7601

**Comment by alexander-akait** [comment]
@yisibl Can you open an issue?

**Comment by yisibl** [comment]
> @yisibl Can you open an issue?

No problem: https://github.com/swc-project/swc/issues/5643

**Pull request merged**